lib/tutorials/index.js

```diff
@@ -125,38 +140,52 @@ module.exports = function makePlugin(dir) {
-                        // Add the demo tag so we can tell later what is a demo
-                        // and what is a full tutorial
-                        tagged[key].tags.push('_demo');
-
-                        for (let i = 0; i < langs.length; i++) {
-                            const lang = langs[i];
-                            const project = tagged[key];
-
-                            // insert tagged project into index
-                            projectDemosIndex[lang][key] = project;
-
-                            // create a new output file for each tagged project
-                            const file = {
-                                title: project.title,
-                                layout: 'tutorial-page.hbs',
-                                tags: project.tags,
-                                mode: '0644',
-                                path: lang + project.url,
-                                locale: lang,
-                                en: true,
-                                description: project.description,
-                                contents: Buffer.from(taggedProjectTemplate(project))
-                            };
-
-                            files[file.path + '/index.html'] = file;
-                        }
+            getTaggedProjects(function (tagged) {
+
+                // add tagged projects into main index
+                for (const key in tagged) {
+
+                    // Add tags from projects to list of tags
+                    for (const tag of tagged[key].tags) {
+                        tutorialTags.add(tag);
                     }
 
+                    // Add the demo tag so we can tell later what is a demo
+                    // and what is a full tutorial
+                    tagged[key].tags.push('_demo');
+
+                    for (let i = 0; i < langs.length; i++) {
+                        const lang = langs[i];
+                        const project = tagged[key];
+
+                        // insert tagged project into index
+                        projectDemosIndex[lang][key] = project;
+
+                        // create a new output file for each tagged project
+                        const file = {
+                            title: project.title,
+                            layout: 'tutorial-page.hbs',
+                            tags: project.tags,
+                            mode: '0644',
+                            path: lang + project.url,
+                            locale: lang,
+                            en: true,
+                            description: project.description,
+                            contents: Buffer.from(taggedProjectTemplate(project))
+                        };
+
+                        files[file.path + '/index.html'] = file;
+                    }
+                }
+
+                // Filter out unwanted tags
+                const unusedTutorialTags = new Set(['monetization', 'tutorial', 'material']);
```

Comment:
Why is monetization and material here? Rather than filter out a specific list, we could check if the tag is in the tutorial tags from the db?

Reply:
Also true, I updated the code so it would ignore a 404 reply from the API when retrieving tags (essentially ignoring tags that don't exist in the db), but would still throw an error for any other status code. It works and I feel like it is justified given the way we're collating the tag lists, but not sure if this is the best solution...